internal/runner/runner.go

```diff
@@ -80,6 +80,8 @@ func (r *Runner) Run() error {
 
 	if r.options.RateLimit > 0 {
 		http.DefaultClient.Transport = utils.NewThrottledTransport(time.Second, r.options.RateLimit, defaultTransport)
+	} else {
+		http.DefaultClient.Transport = defaultTransport
```

Comment:
I think we should avoid unnecessary nesting to make things clearer and potential for slightly better perf.

```go
	http.DefaultClient.Transport = defaultTransport
	if r.options.RateLimit > 0 {
		http.DefaultClient.Transport = utils.NewThrottledTransport(time.Second, r.options.RateLimit, defaultTransport)
	}
```